endpoints/openrtb2/auction.go

```diff
@@ -1339,16 +1313,12 @@ func validateRegs(req *openrtb_ext.RequestWrapper, gpp gpplib.GppContainer) []er
 				WarningCode: errortypes.InvalidPrivacyConsentWarningCode})
 		}
 	}
-	regsExt, err := req.GetRegExt()
-	if err != nil {
-		return append(errL, fmt.Errorf("request.regs.ext is invalid: %v", err))
-	}
-
-	gdpr := regsExt.GetGDPR()
-	if gdpr != nil && *gdpr != 0 && *gdpr != 1 {
-		return append(errL, errors.New("request.regs.ext.gdpr must be either 0 or 1"))
+	if req.BidRequest.Regs.GDPR != nil {
```

Comment:
should the top level be checking for `req.BidRequest.Regs != nil` before accessing the .GDPR field? are we missing test coverage on this case?

Reply:
There's an early return at the top of the function on line 1298 if it is nil:
```
if req == nil || req.BidRequest == nil || req.BidRequest.Regs == nil {
	return nil
}
```

Reply:
Gotcha. Then we don't need this check at all, right? 

```
if req.BidRequest.Regs.GDPR != nil {
  reqGDPR := req.BidRequest.Regs.GDPR
  if reqGDPR != nil && *reqGDPR != 0 && *reqGDPR != 1 {
    return append(errL, errors.New("request.regs.gdpr must be either 0 or 1"))
  }
}
```

After the assignment (which is safe) the first thing you're doing is a repeat of the nil check, so we can get rid of the outer check with no effect.